components/script/dom/htmldialogelement.rs

```diff
@@ -70,6 +70,31 @@ impl HTMLDialogElementMethods for HTMLDialogElement {
         *self.return_value.borrow_mut() = return_value;
     }
 
+    // https://html.spec.whatwg.org/multipage/#dom-dialog-show
+    fn Show(&self) {
+        let element = self.upcast::<Element>();
+
+        // Step 1 TODO: Check is modal flag is false
+        if element.has_attribute(&local_name!("open")) {
+            return;
+        }
+
+        // TODO: Step 2 If this has an open attribute, then throw an "InvalidStateError" DOMException.
+
+        // Step 3
+        element.set_bool_attribute(&local_name!("open"), true);
```

Comment:
Is the spec wrong to say "whose value is the empty string"? Seems like it should indeed be a boolean.

Reply:
The spec isn't wrong, it's just boolean attributes are odd in the spec. A boolean attribute is just about the presence of the attribute so the value is irelevant, I think an attribute has to have a value, so for boolean ones (where you typically omit a value when writing it) they're set to an empty string.